htdocs/core/actions_massactions.inc.php

```diff
@@ -657,6 +657,7 @@
 		} else {
 			// If we want one invoice per order or if there is no first invoice yet for this thirdparty.
 			$objecttmp->socid = $cmd->socid;
+			$objecttmp->fetch_thirdparty();
```

Comment:
I validated PR but modified it after by changing this into 
$objecttmp->thirdparty = $cmd->thirdparty;

Can you confirm it still works as expected ?